url063b9533f6_karlney_tetris_ai_tgz-pJ8-karlney_tetris_cmd_TetrisGameRunnerJ8/src/karlney/tetris/swing/Highscore.java

```diff
@@ -0,0 +1,113 @@
+package karlney.tetris.swing;
+
+import javax.swing.*;
+import java.io.*;
+
+
+
+
+class HighScorePost implements Serializable{
+
+    public int score;
+    public String name;
+
+    public HighScorePost(){}
+
+    public HighScorePost(int s,String n){
+        name=n;
+        score=s;
+    }
+
+    public String toString(){
+        return score+"\t \t"+name;
+    }
+
+}
+
+
+
+
+public class Highscore{
+    static HighScorePost[] scorename = new HighScorePost[10];
+    static String name, outputfile = "highscore.dat";
+    static int place;
+
+
+    public static void readFile(){
+        try{
+            FileInputStream reader = new FileInputStream(outputfile);
+            ObjectInputStream r = new ObjectInputStream(reader);
+            String line = null;
+
+            for (int row=0; row<10; row++){
+                scorename[row]=(HighScorePost)r.readObject();
+                //System.out.println(scorename[row]);
+            }
+
+            r.close();
+        }catch(Exception e){
+            System.out.println(e);
+        }
+    }
+
+
+    public static void updateScore(int score){
+        readFile();
+        if((score>scorename[9].score)){
+            getPlace(score);
+            getName();
+            updateFile(score);
+        }
+        showHighscore();
+    }
+
+
+    public static void getPlace(int score){
+        for(int i=0; i<9; i++){
+            if(score<scorename[i].score){
+            }
+            else{
+                place=i;
+                return;
+            }
+        }
+    }
+
+
+    public static void getName(){
+        name = JOptionPane.showInputDialog("Congratulations, you've made it to the higscore list." + "\n Please enter your name: ");
+        if(name==null || name.length()==0) name="Anonymous";
+    }
+
+    public static void updateFile(int score){
+        try{
+            FileOutputStream writer = new FileOutputStream(outputfile);
+            ObjectOutputStream w = new ObjectOutputStream(writer);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *w*. The resource is referenced at lines 85, 88, 91, 94. The resource is closed at line 96. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *w* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).